gson/src/test/java/com/google/gson/functional/MapTest.java

```diff
@@ -589,8 +589,7 @@ public void testStringKeyDeserialization() {
     Map<String, String> map = new LinkedHashMap<>();
     map.put("2,3", "a");
     map.put("5,7", "b");
-    assertThat(map)
-        .isEqualTo(gson.fromJson(json, new TypeToken<Map<String, String>>() {}.getType()));
+    assertThat(gson.fromJson(json, new TypeToken<Map<String, String>>() {})).isEqualTo(map);
```

Comment:
While you are here, you might as well use the assertions from [`MapSubject`](https://truth.dev/api/latest/com/google/common/truth/MapSubject.html)? So

```java
assertThat(gson.fromJson(...)).containsExactly("2,3", "a", "5,7", "b");
```

Same thing in the next two methods, though not necessarily in `testDeeralizeMapOfMaps`. You could fix the name of that last one, though.

Reply:
Would it be ok if I leave the assertions as they are and not use `MapSubject`? I am currently not completely convinced by `MapSubject.containsExactly` because it makes it a bit difficult to differentiate keys from values, especially here where the keys looks like they might be values (though it is not so bad since this are only 2 entries).

Reply:
Yeah, that's fair.